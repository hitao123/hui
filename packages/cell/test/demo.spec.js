import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Cell from '../index.vue';
import CellDemo from '../demo/index.vue';

describe('Cell component', () => {
  it('should show Cell', () => {
    const warpper = shallowMount(Cell, {
      propsData: {
        title: 'Cell'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show CellDemo', () => {
    expect(renderToString(CellDemo)).toMatchSnapshot();
  });
});
