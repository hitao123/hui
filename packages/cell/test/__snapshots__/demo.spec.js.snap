// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component should show Cell 1`] = `
<div class="h-cell h-hairline">
  <!---->
  <div class="h-cell__title"><span>Cell</span>
    <!---->
  </div>
  <!---->
  <!---->
</div>
`;

exports[`Cell component should show CellDemo 1`] = `
<demo-section>
  <demo-block title="基础用法">
    <h-cell-group>
      <h-cell title="单元格" value="内容"></h-cell>
      <h-cell title="单元格" value="内容" label="描述信息"></h-cell>
    </h-cell-group>
  </demo-block>
  <demo-block title="只设置 value">
    <h-cell-group>
      <h-cell value="内容"></h-cell>
    </h-cell-group>
  </demo-block>
  <demo-block title="设置icon">
    <h-cell-group>
      <h-cell title="单元格" icon="location"></h-cell>
    </h-cell-group>
  </demo-block>
  <demo-block title="展示箭头">
    <h-cell-group>
      <h-cell title="单元格" is-link=""></h-cell>
      <h-cell title="单元格" is-link="" value="内容"></h-cell>
      <h-cell title="单元格" is-link="" arrow-direction="down" value="内容"></h-cell>
    </h-cell-group>
  </demo-block>
  <demo-block title="高级用法">
    <h-cell-group>
      <h-cell value="内容" icon="shop" is-link=""><template><span class="h-cell-text">单元格</span></template></h-cell>
      <h-cell title="单元格" icon="location" is-link=""></h-cell>
      <h-cell title="单元格">
        <h-icon slot="right-icon" name="search" class="h-cell__right-icon"></h-icon>
      </h-cell>
    </h-cell-group>
  </demo-block>
</demo-section>
`;
